pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -87,17 +87,25 @@ var (
 		  oc adm must-gather --image=my/image:tag --source-dir=/pod/directory -- myspecial-command.sh
 	`)
 
+	// According to https://github.com/openshift/enhancements/blob/master/enhancements/oc/must-gather.md#proposal
+	// we can assume /usr/bin/gather is always run in the end, even when a custom command is specified.
+	// That's why we always kill /usr/bin/gather, even though a custom command can be specified.
 	volumeUsageCheckerScript = `
-echo "volume percentage checker started....."
+echo "[disk usage checker] Started"
+target_dir="%s"
+usage_percentage_limit="%d"
 while true; do
-disk_usage=$(du -s "%s" | awk '{print $1}')
-disk_space=$(df -P "%s" | awk 'NR==2 {print $2}')
+disk_usage=$(du -s "$target_dir" | awk '{print $1}')
+disk_space=$(df -P "$target_dir" | awk 'NR==2 {print $2}')
 usage_percentage=$(( (disk_usage * 100) / disk_space ))
-echo "volume usage percentage $usage_percentage"
-if [ "$usage_percentage" -gt "%d" ]; then
-	echo "Disk usage exceeds the volume percentage of %d for mounted directory. Exiting..."
-	# kill gathering process in gather container to prevent disk to use more.
-	pkill --signal SIGKILL -f %s
+echo "[disk usage checker] Volume usage percentage: current = ${usage_percentage} ; allowed = ${usage_percentage_limit}"
+if [ "$usage_percentage" -gt "$usage_percentage_limit" ]; then
+	echo "[disk usage checker] Disk usage exceeds the volume percentage of ${usage_percentage_limit} for mounted directory, terminating..."
+	# Try graceful shutdown first, then force as a fallback. Anchor to avoid matching gather_* helpers.
+	pkill --signal SIGTERM -f "/usr/bin/gather([[:space:]]|$)"
```

Comment:
@tchap I see some examples https://github.com/openshift/release/blob/f113ad4a7bd6c6b5597901b2be6d38186982a0da/ci-operator/step-registry/gather/audit-logs/gather-audit-logs-commands.sh#L31. Instead of checking strict matching, do you think we can kill based on `/usr/bin/gather*`?

Reply:
Sure we can. But I don't find it super robust. Hope it doesn't explode into our face at some point.

Reply:
I find some more test cases for us https://github.com/openshift/release/blob/6f54e610a7785bcfdb5362f29e32c1c6f444668d/ci-operator/step-registry/gather/core-dump/gather-core-dump-commands.sh#L21 

All of them matches the pattern of `/usr/bin/gather_*`. So these are valid `/usr/bin/gather`, `/usr/bin/gather_*`

Reply:
> But I don't find it super robust. Hope it doesn't explode into our face at some point.

I think, if our regex is good enough, this will work

Reply:
Well, just leaving `/usr/bin/gather` is basically a substring match, so that works for `/usr/bin/gather_*` as well. 

Amended.